core/src/main/java/com/novoda/noplayer/exoplayer/CodecSelector.java

```diff
@@ -0,0 +1,47 @@
+package com.novoda.noplayer.exoplayer;
+
+import com.google.android.exoplayer2.mediacodec.MediaCodecInfo;
+import com.google.android.exoplayer2.mediacodec.MediaCodecSelector;
+import com.google.android.exoplayer2.mediacodec.MediaCodecUtil;
+
+/**
+ * CodecSelector that allows overriding of the requiresSecureDecoder to allow
+ * downgrading of security but not upgrading.
+ */
+class CodecSelector implements MediaCodecSelector {
+
+    private final InternalMediaCodecUtil internalMediaCodecUtil;
+    private final boolean overrideSecureDecoder;
+
+    public static CodecSelector newInstance(boolean useSecureCodec) {
```

Comment:
We got confused with this `CodecSelector` too, we decided to invert the logic and do some renaming to make it make sense. 😄  65008cf